### ISSUE TITLE
Raw ENCODE() / DECODE() w/o Base64

### ISSUE THREAD

**Issue by mougino** [RFO-BASIC/Basic#214]
As asked here: http://rfobasic.freeforums.org/encode-and-decode-w-o-base64-conversion-t4180.html

I will propose an enhancement.

Marc, are you ok with the following syntax?
`DECODE$("ENCRYPT_RAW", pwd$, enc$)`

**Comment by mougino** [comment]
My modifications are ready and tested, I am confident the changes are relatively safe.
Marc, tell me how you want we proceed next?

**Comment by jMarcS** [comment]
Thank you, Nicholas. The extension here and the error-reporting in #213 are good changes.

I added one change. Writing the manual is easier if `ENCODE$` sets a `GETERROR$` code for all encodings, not just encryption.

The manual gave me a lot of trouble. The `ENCODE$`/`DECODE$` section is very hard to understand. It's too technical for new programmers but not technical enough to give all the details an expert might need. For instance, it did not say that encrypted data was then converted to Base64. I made it only a little better; it needs a real overhaul.

The difference between `ENCRYPT` and `ENCRYPT_RAW` is not so much the Base64 encoding anyway. `ENCODE` is intended for text strings, especially if they include Unicode (codepoints > 0xFF). `ENCODE_RAW` is intended for binary data in ISO-8859-1 encoded strings. I wrote the manual to emphasize that difference (I used the term "buffer string"), and also tweaked the code and its comments.
